CONTRIBUTING.md

```diff
@@ -43,22 +43,23 @@ We will only accept a pull request for which all tests pass. Make sure the follo
 
 ## Getting started
 
-Please create a new branch from an up to date master on your fork. (Note, urgent hotfixes should be branched off the latest stable release rather than master)
-
 1. Fork the Material-UI repository on GitHub
-2. Clone your fork to your local machine `git clone git@github.com:<yourname>/material-ui.git`
-3. Create a branch `git checkout -b my-topic-branch`
-4. Make your changes, lint, then push to to GitHub with `git push --set-upstream origin my-topic-branch`.
-5. Visit GitHub and make your pull request.
-
-If you have an existing local repository, please update it before you start, to minimise the chance of merge conflicts.
+2. Clone the fork to your local machine and add upstream remote
 ```sh
+git clone git@github.com:<yourname>/material-ui.git
+cd material-ui
 git remote add upstream git@github.com:mui-org/material-ui.git
+```
+
+NOTE: If you already have a cloned Material-UI repository make sure to synchronize it with the upstream one:
+```sh
 git checkout master
 git pull upstream master
-git checkout -b my-topic-branch
-yarn
 ```
+3. Create a branch `git checkout -b my-topic-branch`
+4. Make changes. If you change the code make sure to use proper [coding-style](#coding-style)
+5. Commit and push to your fork with `git push --set-upstream origin my-topic-branch`
```

Comment:
```suggestion
5. Commit and push to your fork with `git push -u origin my-topic-branch`
```

Flag shouldn't be required anyway for the default case (just my opinion) but let's use a friendlier one.